fdbclient/BackupAgentBase.actor.cpp

```diff
@@ -333,23 +353,38 @@ void decodeBackupLogValue(Arena& arena,
 								logValue.param1 = logValue.param1.withPrefix(addPrefix, tempArena);
 								logValue.param2 = logValue.param2.withPrefix(addPrefix, tempArena);
 							}
-							result.push_back_deep(arena, logValue);
-							mutationSize += logValue.expectedSize();
+							result->push_back_deep(*arena, logValue);
+							*mutationSize += logValue.expectedSize();
+						}
+						if (originalLogValue.param1 == logValue.param1 && originalLogValue.param2 == logValue.param2) {
+							encryptedResult->push_back_deep(*arena, encryptedLogValue);
+						} else {
+							encryptedResult->push_back_deep(*arena, Optional<MutationRef>());
 						}
 					}
 				}
 			} else {
 				Version ver = key_version->rangeContaining(logValue.param1).value();
+				bool modified = false;
 				//TraceEvent("ApplyMutation").detail("LogValue", logValue).detail("Version", version).detail("Ver", ver).detail("Apply", version > ver && ver != invalidVersion);
 				if (version > ver && ver != invalidVersion) {
 					if (removePrefix.size()) {
 						logValue.param1 = logValue.param1.removePrefix(removePrefix);
+						modified = true;
 					}
 					if (addPrefix.size()) {
 						logValue.param1 = logValue.param1.withPrefix(addPrefix, tempArena);
+						modified = true;
+					}
+					result->push_back_deep(*arena, logValue);
+					*mutationSize += logValue.expectedSize();
+					// If we did not remove/add prefixes to the mutation then keep the original encrypted mutation so we
```

Comment:
The comment seems bit misleading, unless add/remove prefixes crosses encryption boundaries, from encryption perspective it should be a NOP?

Reply:
Well the idea is that if we don't add or remove any prefix (i.e don't change the mutation) then we can bypass encryption in the commitproxy. However if we do add or remove a prefix this changes the mutation so we cannot simply use the old mutation

Reply:
For example in a cluster where tenants are disabled we may change the prefix and even though this doesn't cross tenant boundaries we still need to re-encrypt as the mutation has changed